allennlp/training/callback_trainer.py

```diff
@@ -117,6 +117,15 @@ def __init__(self,
         self.epoch_number = 0
         self.batch_grad_norm: Optional[float] = None
 
+        # If you're doing something like gradient accumulation, you might want
+        # to modify this during training to treat multiple forward / backward
+        # passes as a single "batch".
+        self.is_start_of_batch = True
```

Comment:
These variable names are still pretty obnoxious because they can both be true simultaneously... maybe something like `should_run_epoch_start` and `should_wait_for_grad_step` or something?